### ISSUE TITLE
Incorrect params passed from Vault::Rails::with_retries to Vault::Client#with_retries

### ISSUE THREAD

**Issue by tvon** [hashicorp/vault-rails#36]
### TLDR

vault-ruby expects `with_retries(Exception1, Exception2, options_hash)` whereas vault-rails is passing `with_retries([Exception1, Exception2], options_hash)`.  This results in vault-ruby expanding it's rescue to `rescue [[Exception1, Exception2]] => e` which will throw a `TypeError`, but only when there is an exception to rescue (so the happy path does not expose it).

This can probably be fixed by splatting the exceptions argument in vault-rails.

### Long version

`Vault::Rails::with_retries` passes `exceptions` to `Vault::Client#with_retries` [as a distinct array](https://github.com/hashicorp/vault-rails/blob/84058241be1886e66aac5e90aa28c8cad7a82d5e/lib/vault/rails.rb#L204), e.g.:

```ruby
exceptions = [Vault::HTTPConnectionError, Vault::HTTPServerError]
client.with_retries(exceptions, options)

# expanded:
client.with_retries([Vault::HTTPConnectionError, Vault::HTTPServerError], options)
```

What `Vault::Client#with_retries` expects is each exception as a parameter, e.g.:
```ruby
client.with_retries(Vault::HTTPConnectionError, Vault::HTTPServerError, options)
```

As it is currently written, `Vault::Client#with_retries` [ends up with a nested array in the rescue clause](https://github.com/hashicorp/vault-ruby/blob/35a14f775810ac26f6da0070f60ccf2faab6e84c/lib/vault/client.rb#L384):

```ruby
def with_retries(*rescued, &block)
  # "rescued" evaluates to a nested array:
  #   [[Vault::HTTPConnectionError, Vault::HTTPServerError], options]
  # "options" is popped off of that, so "rescued" ends up as [[Vault::HTTPConnectionError, Vault::HTTPServerError]]
  # later:
  rescue *retries => e
  # ...
end

# the rescue expands to:
rescue [[Vault::HTTPConnectionError, Vault::HTTPServerError]] => e
# The above is a TypeError, the correct syntax is:
rescue [Vault::HTTPConnectionError, Vault::HTTPServerError] => e
```

This is only exposed if the rescue is executed, it will not show up through the happy path.  The result is that if you are investigating vault-rails and there is some configuration issue (vault not running, incorrect URL, incorrect token) instead of a clear error you get `TypeError: class or module required for rescue clause`

I believe this can be resolved by [changing this line](https://github.com/hashicorp/vault-rails/blob/84058241be1886e66aac5e90aa28c8cad7a82d5e/lib/vault/rails.rb#L211):

```diff
- client.with_retries(exceptions, options) do |i, e|
+ client.with_retries(*exceptions, options) do |i, e|
```

But I am unable to PR/test it at the moment.
